include/psp2/usbd.h

```diff
@@ -7,89 +7,505 @@
+    SCE_USBD_DESCRIPTOR_REPORT            = 0x22     // bDescriptorType for an HID report descriptor.
+} SceUsbdDescriptorType;
+
+typedef struct SceUsbdDeviceDescriptor {
+    uint8_t bLength;
+    uint8_t bDescriptorType;
+    uint16_t bcdUSB;
+    uint8_t bDeviceClass;
+    uint8_t bDeviceSubclass;
+    uint8_t bDeviceProtocol;
+    uint8_t bMaxPacketSize0;
+    uint16_t idVendor;
+    uint16_t idProduct;
+    uint16_t bcdDevice;
+    uint8_t iManufacturer;
+    uint8_t iProduct;
+    uint8_t iSerialNumber;
+    uint8_t bNumConfigurations;
+} SceUsbdDeviceDescriptor;
+
+typedef struct SceUsbdConfigurationDescriptor {
+    uint8_t bLength;
+    uint8_t bDescriptorType;
+    uint16_t wTotalLength;
+    uint8_t bNumInterfaces;
+    uint8_t bConfigurationValue;
+    uint8_t iConfiguration;
+    uint8_t bmAttributes;
+    uint8_t MaxPower;
+} SceUsbdConfigurationDescriptor;
+
+#define SCE_USBD_CONFIGURATION_RESERVED_ZERO         0x1f
+#define SCE_USBD_CONFIGURATION_REMOTE_WAKEUP         0x20
+#define SCE_USBD_CONFIGURATION_SELF_POWERED          0x40
+#define SCE_USBD_CONFIGURATION_RESERVED_ONE          0x80
+
+typedef struct SceUsbdInterfaceDescriptor {
+    uint8_t bLength;
+    uint8_t bDescriptorType;
+    uint8_t bInterfaceNumber;
+    uint8_t bAlternateSetting;
+    uint8_t bNumEndpoints;
+    uint8_t bInterfaceClass;
+    uint8_t bInterfaceSubclass;
+    uint8_t bInterfaceProtocol;
+    uint8_t iInterface;
+} SceUsbdInterfaceDescriptor;
+
+typedef struct SceUsbdEndpointDescriptor {
+    uint8_t bLength;
+    uint8_t bDescriptorType;
+    uint8_t bEndpointAddress;
+    uint8_t bmAttributes;
+    uint16_t wMaxPacketSize;
+    uint8_t bInterval;
+} SceUsbdEndpointDescriptor;
+
+/* bmAttributes */
+#define SCE_USBD_ENDPOINT_TRANSFER_TYPE_BITS         0x03
+#define SCE_USBD_ENDPOINT_TRANSFER_TYPE_SHIFT        0
+#define SCE_USBD_ENDPOINT_TRANSFER_TYPE_CONTROL      0x00
+
+#define SCE_USBD_ENDPOINT_TRANSFER_TYPE_ISOCHRONOUS  0x01
+#define SCE_USBD_ENDPOINT_TRANSFER_TYPE_BULK         0x02
+#define SCE_USBD_ENDPOINT_TRANSFER_TYPE_INTERRUPT    0x03
+
+/* bEndpointAddress */
+#define SCE_USBD_ENDPOINT_NUMBER_BITS                0x1f
+#define SCE_USBD_ENDPOINT_NUMBER_SHIFT               0
+#define SCE_USBD_ENDPOINT_DIRECTION_BITS             0x80
+#define SCE_USBD_ENDPOINT_DIRECTION_SHIFT            7
+#define SCE_USBD_ENDPOINT_DIRECTION_OUT              0x00
+#define SCE_USBD_ENDPOINT_DIRECTION_IN               0x80
+
+typedef struct SceUsbdStringDescriptor {
+    uint8_t bLength;
+    uint8_t bDescriptorType;
+    uint8_t bString[0];
+} SceUsbdStringDescriptor;
+
+typedef struct SceUsbdHidSubDescriptorInfo {
+    uint8_t bDescriptorType;
+    uint8_t wDescriptorLength0;
+    uint8_t wDescriptorLength1;
+} SceUsbdHidSubDescriptorInfo;
+
+typedef struct SceUsbdHidDescriptor {
+    uint8_t bLength;
+    uint8_t bDescriptorType;
+    uint8_t bcdHID0;
+    uint8_t bcdHID1;
+    uint8_t bCountryCode;
+    uint8_t bNumDescriptors;  /* SubDescriptor count */
```

Comment:
can you change these field variable names to `snake_case`?

Reply:
No. That's how they names on every other system. Renaming them just makes porting harder.